### PR TITLE
Update GazeProvider to use SpatialPointerPose.Head

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

This PR adds a new gaze provider interface, `IMixedRealityGazeProviderWithOverride`, to prevent a breaking change to the existing gaze provider interface. This new interface allows a data provider to override the gaze data with platform-specific values. In this case, the data comes from `SpatialPointerPose.Head`.

I also reorganized `GazeProvider` a bit to group implementations by interface.

TODO BEFORE MERGING (but wanted to get review going):

* [x] Docs

## Changes

- Fixes: #6436

### CODE REVIEW

File: Assets/MRTK/Services/InputSystem/GazeProvider.cs [hunk 1]
```diff
@@ -75,16 +75,15 @@ public class GazeProvider :
...
+
         /// <inheritdoc />
         public bool Enabled
         {
             get { return enabled; }
             set { enabled = value; }
         }
 
-        /// <inheritdoc />
-        public bool IsEyeTrackingEnabled { get; set; }
```
Comment (keveleigh) [inline Assets/MRTK/Services/InputSystem/GazeProvider.cs]: This (along with other eye tracking properties that appear removed) isn't removed but moved to the eye gaze implementation region lower down.

File: Assets/MRTK/Core/Interfaces/InputSystem/IMixedRealityGazeProviderWithOverride.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Adds ability to override head gaze on a gaze provider.
+    /// </summary>
+    public interface IMixedRealityGazeProviderWithOverride
```
Comment (davidkline-ms) [inline Assets/MRTK/Core/Interfaces/InputSystem/IMixedRealityGazeProviderWithOverride.cs]: nit: maybe IMixedRealityGazeProviderHeadOverride?

Comment (keveleigh) [reply Assets/MRTK/Core/Interfaces/InputSystem/IMixedRealityGazeProviderWithOverride.cs]: Yeah, I think I like yours better! Changed this a couple times and was never fully happy with it. Thanks!

Comment (keveleigh) [reply Assets/MRTK/Core/Interfaces/InputSystem/IMixedRealityGazeProviderWithOverride.cs]: Renamed

**Pull request merged**